db/mysql/dao/gateway.go

```diff
@@ -313,6 +313,66 @@ func (h *HTTPRuleDaoImpl) ListByComponentIDs(componentIDs []string) ([]*model.HT
 	return rules, nil
 }
 
+// HTTPRuleRewriteDaoTmpl is a implementation of HTTPRuleRewriteDao
+type HTTPRuleRewriteDaoTmpl struct {
+	DB *gorm.DB
+}
+
+//AddModel -
+func (h *HTTPRuleRewriteDaoTmpl) AddModel(mo model.Interface) error {
+	httpRuleRewrite, ok := mo.(*model.HTTPRuleRewrite)
+	if !ok {
+		return fmt.Errorf("can't not convert %s to *model.HTTPRuleRewrite", reflect.TypeOf(mo).String())
+	}
+	var oldHTTPRuleRewrite model.HTTPRuleRewrite
+	if ok := h.DB.Where("uuid = ?", httpRuleRewrite.UUID).Find(&oldHTTPRuleRewrite).RecordNotFound(); ok {
+		if err := h.DB.Create(httpRuleRewrite).Error; err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("HTTPRuleRewrite already exists based on uuid(%s)", httpRuleRewrite.UUID)
+	}
+	return nil
+}
+
+//UpdateModel -
+func (h *HTTPRuleRewriteDaoTmpl) UpdateModel(mo model.Interface) error {
+	hr, ok := mo.(*model.HTTPRuleRewrite)
+	if !ok {
+		return fmt.Errorf("failed to convert %s to *model.HTTPRuleRewrite", reflect.TypeOf(mo).String())
+	}
+	return h.DB.Save(hr).Error
+}
+
+// CreateOrUpdateHTTPRuleRewriteInBatch -
+func (h *HTTPRuleRewriteDaoTmpl) CreateOrUpdateHTTPRuleRewriteInBatch(httpRuleRewrites []*model.HTTPRuleRewrite) error {
+	var objects []interface{}
+	for _, httpRuleRewrites := range httpRuleRewrites {
```

Comment:
The traversed variables are the same as the slice name, which is easy to be confused